libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2078,6 +2086,97 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 	}
 }
 
+void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCall)
+{
+	vector<ASTPointer<Expression const>> const& arguments = _functionCall.arguments();
+
+	// Expecting first argument to be the function pointer and second to be a tuple.
+	if (arguments.size() != 2)
+	{
+		m_errorReporter.typeError(
+			6219_error,
+			_functionCall.location(),
+			"Expected two arguments: a function pointer followed by a tuple."
+		);
+		return;
+	}
+
+	auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front()));
+
+	if (!functionPointerType)
+	{
+		m_errorReporter.typeError(
+			5511_error,
+			arguments.front()->location(),
+			"Expected first argument to be a function pointer, not \"" +
+			type(*arguments.front())->canonicalName() +
+			"\"."
+		);
+		return;
+	}
+
+	if (functionPointerType->kind() != FunctionType::Kind::External)
+	{
+		string msg = "Function must be \"public\" or \"external\".";
+		SecondarySourceLocation ssl{};
+
+		if (functionPointerType->hasDeclaration())
+		{
+			ssl.append("Function is declared here:", functionPointerType->declaration().location());
+			if (functionPointerType->declaration().scope() == m_currentContract)
+				msg += " Did you forget to prefix \"this.\"?";
```

Comment:
```suggestion
				msg += " Did you forget to prefix it with \"this.\"?";
```

Reply:
I mean, prefix is also a verb? I think both is valid.

Reply:
I also like just prefix

Reply:
Both are valid but I think that the meaning is different:
- `prefix "this."` = `add a prefix to "this."`. I.e. `this.x` -> `prefix.this.x`.
- `prefix it with "this."` = `add "this." as a prefix to it`. I.e. `x` -> `this.x`.

I mean, it's will probably be understood in context but technically the first version is wrong here.

Reply:
ok, maybe it's my interpretation that's wrong after all: https://www.merriam-webster.com/dictionary/prefix:

> 1. \ (ˌ)prē-​ˈfiks: to fix or appoint beforehand
> 2. \ ˈprē-​ˌfiks, prē-​ˈfiks \ [partly from prefix entry 2] : to place in front
>     especially : to add as a prefix prefix a syllable to a word 
>     // prefix a syllable to a word

 Weird. I always understood "prefix something" as "add prefix to something".